tests/image_test.py

```diff
@@ -1358,6 +1358,10 @@ def test_validate_filename(self, tmpdir):
         assert not image.validate_filename(str(filename), valid_extensions)
         assert not image.validate_filename('some_test_file.png', valid_extensions)
 
+    def test_iterator_empty_directory(self):
+        iterator = image.Iterator(0, 0, False, 0)
+        next(iterator.index_generator)
```

Comment:
test that there a no samples returned by the iterator., add also test with a different batch size